### PR TITLE
Ial issue 16 bypassed automation

### PR DESCRIPTION
**Pull request by merktj**
Attempts to fix issue #16 .
Implements new function to check whether envelopes are bypassed and introduces new if-case to check for that.

### CODE REVIEW

File: reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp [hunk 1]
```diff
@@ -356,6 +364,26 @@ std::optional<double> EarVstExportSources::getValueFor(std::shared_ptr<admplug::
     return std::optional<double>();
 }
 
+bool EarVstExportSources::getEnvelopeBypassed(TrackEnvelope* env, ReaperAPI const& api)
+{
+    bool envBypassed = false;
+    char chunk[1024]; // A new envelope should only be about 100 bytes
```
Comment (rsjtaylor) [inline reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp:370]: Is this assumption valid? https://forums.cockos.com/showthread.php?p=2142251 suggests envelope points are stored in the state chunk and something like the turning forest (S3A dataset available here https://cvssp.org/data/s3a/public/radiodrama_register.php) have a lot of object automation points (thousands?)

File: reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp [hunk 2]
```diff
@@ -356,6 +364,26 @@ std::optional<double> EarVstExportSources::getValueFor(std::shared_ptr<admplug::
...
+    bool getRes = api.GetEnvelopeStateChunk(env, chunk, 1024, false);
+    if (getRes) {
+        std::istringstream chunkSs(chunk);
+        std::string line;
+        while (std::getline(chunkSs, line)) {
+            auto activePos = line.rfind("ACT ", 0);
+            if (activePos != std::string::npos) {
+                auto active = static_cast<int>(line.at(activePos + 4)) - 48;
```
Comment (rsjtaylor) [inline reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp:378]: Potentially two issues here. 
First, in the (unlikely) event the chunk got truncated so the returned string ended with "ACT " (see link in comment about size assumption) I think the `rfind()` would succeed but the `.at()` would throw an out of range exception as std::strings don't store the terminating char.
Second, is there a reason for the cast / ascii arithmetic or could you just do 
```c++
envBypassed = line.at(activePos + 4) == '0';
```

File: reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp [hunk 3]
```diff
@@ -193,8 +193,16 @@ void EarVstExportSources::generateAdmAndChna(ReaperAPI const& api)
                 auto admParameter = (AdmParameter)admParameterIndex;
                 auto param = pluginSuite->getParameterFor(admParameter);
                 auto env = getEnvelopeFor(pluginSuite, pluginInst.get(), admParameter, api);
-
-                if(param && env) {
+                
+                if (getEnvelopeBypassed(env, api)) { 
+                    // We have an envelope, but it is bypassed
```
Comment (firthm01) [inline reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp:198]: I don't think we can assume there is an envelope at this point - env could be nullptr

Comment (firthm01) [reply reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp:198]: Looking at `getEnvelopeBypassed` - does `api.GetEnvelopeStateChunk` return false if `env` is `nullptr`? in which case I can see how `getEnvelopeBypassed` would be false if there is no envelope.

Comment (merktj) [reply reaper-adm-extension/src/reaper_adm/exportaction_admsource-earvst.cpp:198]: Thanks for the review. If `env == nullptr`, `getEnvelopeBypassed` returns `false`. But I see that this wasn't a very elegant way to handle it. I've extended the function with an other if-case to handle the  `env == nullptr` case now.

**Pull request merged**